demo/src/app/shared/markdown/code.component.ts

```diff
@@ -0,0 +1,23 @@
+import {Component, OnInit, Input, ChangeDetectionStrategy, ElementRef} from '@angular/core';
+import {MarkdownService} from './markdown.service';
+
+import * as prism from 'prismjs';
+
+@Component({
+  selector: 'ngbd-code',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: ``
+})
+export class NgbdCodeComponent implements OnInit {
+
+  @Input() code = '';
+  @Input() lang = '';
+
+  constructor(private service: MarkdownService, private el: ElementRef) { }
+
+  ngOnInit() {
+    const code = `\`\`\`${this.lang}\n${this.code.trim()}\n\`\`\``;
+    this.el.nativeElement.innerHTML = this.service.toHTML(code);
```

Comment:
Not sure why do we pass code through markdown service?

Reply:
Oh, wow, got a little excited there playing with markdown. Will remove markdown completely from this PR and leave only code highlighting component